lib/functions/name/name.ts

```diff
@@ -0,0 +1,46 @@
+interface FormatDisplayNameInput {
+  /** Full name of person or company */
+  fullName: string;
+  /** Entity type: Whether this is a person or a company */
+  type: 'person' | 'company';
+  /** For persons: switch order of first/last name */
+  reverseNameOrder?: boolean;
+}
+
+/**
+ * Formats a display name for a person or company by applying title casing,
+ * normalizing hyphenated words, and (for companies) uppercasing known legal
+ * entity suffixes.
+ *
+ * For persons, you can optionally reorder the name by moving the first word
+ * (often a surname) to the end.
+ *
+ * @param {FormatDisplayNameInput} input - The input object with full name, entity type, and options.
+ * @returns {string} The formatted display name.
+ *
+ */
+export const formatDisplayName = ({ fullName, type, reverseNameOrder }: FormatDisplayNameInput): string => {
+  if (!fullName) return '';
+
+  const legalEntityTypes = ['enk', 'as', 'da', 'sa'];
+  let words = fullName
+    .toLowerCase()
+    .split(' ')
+    .map((word) => {
+      if (type === 'company' && legalEntityTypes.includes(word)) {
```

Comment:
Should this check only be at the end?

Words like 'as' could easily appear in the middle of a name without being a reference to the type.
For example, the fictional organization "Work as Friends AS":

So maybe the legalEntityTypes check should only happen at the last word of the name?